plugin/attributes.go

```diff
@@ -20,22 +34,27 @@ type CatalogAttributes struct {
 func getCatalogAttributes(in *structpb.Struct) (*CatalogAttributes, error) {
 	unknownFields := structFields(in)
 	result := new(CatalogAttributes)
+	badFields := make(map[string]string)
 
 	var err error
 	result.Region, err = getStringValue(in, constRegion, true)
 	if err != nil {
-		return nil, err
+		badFields["attributes.region"] = err.Error()
```

Comment:
Do the same thing here as you did in other places.  Do this for all places where there is a const string being used to look up the values.  This is good because if this field name needs to be changed we have a single place to change it, it ensures consistency in naming across locations, and it helps avoid typos.

```suggestion
		badFields[fmt.Sprintf("attributes.%s", constRegion)] = err.Error()
```